test/e2e/run-compat-matrix.sh

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env bash
+# Copyright 2024 The KubeStellar Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# E2E compatibility matrix test for KubeFlex CLI and KubeStellar core-chart.
+# Iterates over all KubeFlex CLI versions >= min required, runs E2E tests for each.
+
+set -euo pipefail
+set -x
+
+# Get min required KubeFlex CLI version from check_pre_req.sh
+MIN_KFLEX_VERSION=$(grep -oE "Kubeflex version: v[0-9]+\.[0-9]+\.[0-9]+" "$(dirname "$0")/../../scripts/check_pre_req.sh" | head -1 | grep -oE "[0-9]+\.[0-9]+\.[0-9]+")
+
+# Preflight: ensure required tools are present; if not, skip gracefully in CI
+required=(kubectl helm kind docker curl git)
+for t in "${required[@]}"; do
+  if ! command -v "$t" >/dev/null 2>&1; then
+    echo "compat-matrix: required tool '$t' not found; skipping (exit 0)" >&2
+    exit 0
+  fi
+done
+
+# Fetch available KubeFlex CLI versions from git tags (public repo)
+function fetch_kflex_versions() {
+  # Outputs plain semver (no leading 'v'), sorted ascending
+  git ls-remote --tags https://github.com/kubestellar/kubeflex.git \
+    | awk '{print $2}' \
+    | sed -n 's#refs/tags/v\([0-9]\+\.[0-9]\+\.[0-9]\+\)$#\1#p' \
```

Comment:
When I tested this script, it failed with a complaint about no versions found.

```console
bash-3.2$ sed -n 's#refs/tags/v\([0-9.]*\)$#\1#p'
refs/tags/v0.7.2
refs/tags/v0.8.0
0.7.2
0.8.0
```

.. but

```console
bash-3.2$ sed -n 's#refs/tags/v\([0-9.]\+\)$#\1#p'
refs/tags/v0.7.2
refs/tags/v0.8.0
```